src/components/GameboardOnline/GameboardOnline.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+import { getAuth } from "firebase/auth";
+import { gameIsOver, updateGameOnline } from "../../services/game.service";
+import { PlayCard } from "../PlayCard/PlayCard";
+
+const turnIsAllowed = (turn) => turn < 2;
+
+const endOfGame = (board) => {
+  return board.every((card) => card.cleared);
+};
+
+const getNextPlayer = ({ players, currentPlayer }) => {
+  const playerArray = Object.keys(players);
+  const currentPlayerIndex = playerArray.indexOf(currentPlayer);
+  const nextPlayerIndex = (currentPlayerIndex + 1) % playerArray.length;
+  return playerArray[nextPlayerIndex];
+};
+
+const revealCard = ({ cardID, board }) => {
+  const newBoard = board.map((card) => {
+    if (cardID === card.id) {
+      return {
+        ...card,
+        reveal: true,
+      };
+    }
+    return card;
+  });
+  return newBoard;
+};
+
+const unrevealCards = (board) => {
+  return board.map((card) => (card.reveal ? { ...card, reveal: false } : card));
+};
+
+const revealedCardsMatch = (board) => {
+  const revealedCards = board.filter((card) => card.reveal);
+  return (
+    revealedCards.map((card) => card.id).reduce((acc, cur) => acc + cur) === 0
+  );
+};
+const clearCards = (board) => {
+  return board.map((card) => {
+    return card.reveal ? { ...card, cleared: true, reveal: false } : card;
+  });
+};
+
+export const GameboardOnline = ({ game, ...props }) => {
+  const { id: gameID, board, turn } = game;
+  const userID = getAuth().currentUser.uid;
+  const itsMyTurn = userID === game.currentPlayer;
+  const endOfTurn = turn === 2;
+
+  const handleEndOfTurn = () => {
+    if (revealedCardsMatch(board)) {
+      const newScore = game.score[userID] + 1;
+      const newBoard = clearCards(board);
+      const updates = {
+        turn: 0,
+        board: newBoard,
+        [`score/${userID}`]: newScore,
+      };
+      if (endOfGame(newBoard)) updates.state = "done";
+      updateGameOnline({
+        gameID,
+        updates,
+      });
+    } else {
+      const newBoard = unrevealCards(board);
+      const nextPlayer = getNextPlayer(game);
+      updateGameOnline({
+        gameID,
+        updates: {
+          turn: 0,
+          board: newBoard,
+          currentPlayer: nextPlayer,
+        },
+      });
+    }
+
+    // if EOG handle EOG
```

Comment:
EOG heißt end of game? :)

Reply:
Setzt hier mal ein //TODO: .... damit wir das mit dem add on gelistet bekommen

Reply:
ups, das war nur meine Schreibhilfe. Versuche bei so komplizierteren Geschichten immer erstmal alles in Text zu fassen und anschließend in code umzuwandeln. Habs gelöscht.